firmware/controllers/actuators/idle_thread.h

```diff
@@ -58,6 +58,9 @@ class IdleController : public IIdleController {
 	// CLOSED LOOP CORRECTION
 	float getClosedLoop(IIdleController::Phase phase, float tpsPos, int rpm, int targetRpm) override;
 
+	void onConfigurationChange(engine_configuration_s const * previousConfig) final;
```

Comment:
`override`?

Reply:
eh, whatever.  Importantly, it needs to have one or the other to detect changes in the underlying API.  That's the only reason EngineModule needs virtual functions, otherwise we could rely on duck typing.

Because of the use of generic lambdas, it doesn't actually affect performance - the compiler knows the exact type and can call the functions directly withing using the vtable.

Reply:
Yes, I know that it will never actually hit the vtable.

Why does it need the base `EngineModule` at all? Why not rely on duck typing to enforce presence of any functions we try to call?

Reply:
It's convenient to not have to declare dummies for each callback you don't care about.